From 5a4f2ddb0a6c656b425a3caa41f5af766ed5f44b Mon Sep 17 00:00:00 2001
From: Alex Kiernan <alex.kiernan@gmail.com>
Date: Mon, 9 Sep 2019 14:19:18 +0100
Subject: [PATCH] macros: Add TEMP_FAILURE_RETRY for musl

TEMP_FAILURE_RETRY is glibc specific, add a definition for musl.

See https://github.com/ostreedev/ostree/issues/731

Signed-off-by: Leon Woestenberg <leon@sidebranch.com>
Signed-off-by: Alex Kiernan <alex.kiernan@gmail.com>
Upstream-Status: Accepted [https://gitlab.gnome.org/GNOME/libglnx/commit/15c06468804014fc49f6f340132930703f919b19]
---
 glnx-macros.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/libglnx.glnx-macros.h b/libglnx/glnx-macros.h
index 6d8aca93f8a3..700fc75ceed0 100644
--- a/libglnx.glnx-macros.h
+++ b/libglnx/glnx-macros.h
@@ -31,6 +31,16 @@ G_BEGIN_DECLS
 /* All of these are for C only. */
 #ifndef __GI_SCANNER__
 
+/* fixes builds against musl, taken from glibc unistd.h */
+#ifndef TEMP_FAILURE_RETRY
+#define TEMP_FAILURE_RETRY(expression) \
+  (__extension__                                                              \
+    ({ long int __result;                                                     \
+       do __result = (long int) (expression);                                 \
+       while (__result == -1L && errno == EINTR);                             \
+       __result; }))
+#endif
+
 /* Taken from https://github.com/systemd/systemd/src/basic/string-util.h
  * at revision v228-666-gcf6c8c4
  */
-- 
2.17.1

